src/sql/workbench/contrib/notebook/browser/notebookActions.ts

```diff
@@ -487,10 +487,6 @@ export class AttachToDropdown extends SelectBox {
 
 			let attachToConnections = this.values;
 			if (!connection) {
-				// If there is no connection, we should choose the previous connection,
-				// which will always be the first item in the list. Either "Select Connection"
-				// or a real connection name
-				this.select(0);
 				return false;
```

Comment:
Where is select defined?

Reply:
Good question, I am not sure why we actually had this as select was supposed to reference the AttachTo dropdown items. 
this.select is defined on the `(method) SelectBox.select(index: number): void` method. 
However, there is no reason to have this inside the condition of the (!connection). 
The AttachTo dropdown list will always have two options:
- Select Connection
- Change Connection

Reply:
Please look at the line history view to get context why this was added in the first place.

Reply:
I did, I should've rephrased as I am just quite puzzled as to why this was only found now. This error must have been present in past releases as well since that change was implemented a year ago. But it also makes sense since there has been quite a lot of change to notebooks and connections (specifically due to Kusto).

Reply:
I'm confused. The bug is a new regression from the previous stable release.

Stable:
- Create new notebook
- Create new cell
- Press F5 (connection dialog opens)
- Press escape (connection dialog closes)

Now, in RC, I have to press escape twice. What changed recently?

Reply:
So I stepped through the entire process from new cell to pressing cancel on the Connection Dialog. I also looked at the git lens the entire time. 
Files hit in order upon hitting F5 (that are relevant): 
- notebookComponent.ts (runCell)
- notebookModel (requestConnection)
- notebookActions (openConnectionDialog)
- cell.ts (runCell)

Files hit upon hitting Cancel on Connection Dialog:
- connectionDialogWidget (cancel method)
- notebookComponent (runCell) 
- notebookModel (requestConnection)
- notebookAction (openConnectionDialog - !connection case)

At this point there is no file or line that was changed recently (the closest one was the notebookComponent runCell that had a VScode Merge 4 months ago) 

I took all this information and clearly saw that this.select reopens the connectionDialog. As a result, I removed it and saw it worked as expected now.

Reply:
Looks like this is caused by changes to how SelectBox fires events when calling `select()`: https://github.com/microsoft/azuredatastudio/commit/63536eba9fe498db863469646dc4780a1eb5dd62

@Charles-Gagnon has the context. Let me think about the fix here too.

Reply:
Maybe something like this? https://github.com/microsoft/azuredatastudio/compare/chlafren/playing? Trying to think if it makes more sense to make this configurable or pursue other options.

Any thoughts here @Charles-Gagnon?

Reply:
I'll take a look.